src/sqlstorage.cc

```diff
@@ -347,7 +315,17 @@ bool SQLStorage::loadTlsPkey(std::string* pkey) {
 void SQLStorage::storeMetadata(const Uptane::MetaPack& metadata) {
   SQLite3Guard db(config_.sqldb_path.c_str());
 
-  clearMetadata();
+  if (db.get_rc() != SQLITE_OK) {
+    LOG_ERROR << "Can't open database: " << db.errmsg();
+    return;
+  }
+
+  if (!db.beginTransaction()) return;
+
+  if (db.exec("DELETE FROM meta;", NULL, NULL) != SQLITE_OK) {
+    LOG_ERROR << "Can't clear meta: " << db.errmsg();
+    return;
```

Comment:
What happens in a place like this when there is an early return? Does the state created by the beginTransaction get cleared despite that the commitTransaction never gets called?

Otherwise I think this looks fine.

Reply:
When `SQLite3Guard` is destructed, it calls the destructor of `handle_` which calls `sqlite3_close()`.

> If an sqlite3 object is destroyed while a transaction is open, the transaction is automatically rolled back.

https://www.sqlite.org/c3ref/close.html

`SQLiteStatement` are freed similarly with `sqlite3_finalize`.

I'll add some comments to clarify this.